src/mlpack/tests/CMakeLists.txt

```diff
@@ -248,3 +268,4 @@ endforeach()
 
```

Comment:
(I commented here because this is as close as I can get to the relevant code...)

I think we need to call `add_test()` for the new test that you added, like above:

```
    add_test(NAME ${test} COMMAND mlpack_test -t ${test} WORKING_DIRECTORY
      ${CMAKE_BINARY_DIR})
```

Otherwise Azure and the other CI systems won't actually run the new test.  I think that you could just keep it simple:

```
add_test(NAME "catch_test" COMMAND mlpack_catch_test WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
```
I'm not sure if that command is exactly right, but in any case, when you look at the Azure output for the testing stage, you should be able to find the `mlpack_catch_test` run in there somewhere:

https://dev.azure.com/mlpack/mlpack/_build/results?buildId=2807&view=logs&jobId=204580d5-7a27-510d-d602-9122b08edba8&j=204580d5-7a27-510d-d602-9122b08edba8&t=e7f486a7-4dfd-5e54-539b-fead054ff497